.evergreen/scripts/compile-windows.sh

```diff
@@ -104,11 +104,13 @@ if [ "${COMPILE_LIBMONGOCRYPT}" = "ON" ]; then
   git clone --depth=1 https://github.com/mongodb/libmongocrypt --branch 1.7.0-alpha1
   mkdir libmongocrypt/cmake-build
   pushd libmongocrypt/cmake-build
-  "${cmake_binary}" -G "${CC}" "-DCMAKE_PREFIX_PATH=${install_dir}/lib/cmake" -DENABLE_SHARED_BSON=ON -DCMAKE_INSTALL_PREFIX="${install_dir}" ../
+  "${cmake_binary}" -G "${CC}" "-DCMAKE_PREFIX_PATH=$(to_windows_path "${install_dir}")/lib/cmake" -DENABLE_SHARED_BSON=ON -DCMAKE_INSTALL_PREFIX="$(to_windows_path "${install_dir}")" ../
```

Comment:
```suggestion
  "${cmake_binary}" -G "${CC}" "-DCMAKE_PREFIX_PATH=$(to_windows_path "${install_dir}/lib/cmake")" -DENABLE_SHARED_BSON=ON -DCMAKE_INSTALL_PREFIX="$(to_windows_path "${install_dir}")" ../
```
Ensure the `/lib/cmake` is part of converted Windows path.